### PR TITLE
Doc aliases for C & renamed functions

### PR DESCRIPTION
**Pull request by fengalin**
(At least part of) https://github.com/gtk-rs/gir/issues/909.

Was handled while preparing https://github.com/gtk-rs/gir/issues/963.

### CODE REVIEW

File: src/codegen/function.rs [hunk 1]
```diff
@@ -74,6 +75,12 @@ pub fn generate(
     version_condition(w, env, analysis.version, commented, indent)?;
     not_version_condition(w, analysis.not_version, commented, indent)?;
     doc_hidden(w, analysis.doc_hidden, comment_prefix, indent)?;
+    if !in_trait || only_declaration {
+        doc_alias(w, &analysis.glib_name, comment_prefix, indent)?;
+        if analysis.new_name.is_some() {
+            doc_alias(w, &analysis.name, comment_prefix, indent)?;
```
Comment (GuillaumeGomez) [inline src/codegen/function.rs:81]: An example for this second doc_alias:

```rust
#[doc(alias = "gdk_pixbuf_new_from_bytes")]
#[doc(alias = "new_from_bytes")]
pub fn from_bytes() {}
```

I think there will be a lot of of equivalents for `new_from_bytes` and I'm sure there is much point into having it...

Comment (fengalin) [reply src/codegen/function.rs:81]: Yes indeed. See new version.

**Comment by GuillaumeGomez** [comment]
Looks much better, thanks!

File: src/codegen/function.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/codegen/function.rs:81]: Looks much better, thanks!

**Comment by sdroege** [comment]
Let's get this in after https://github.com/gtk-rs/gir/pull/991 :) One step at a time

File: src/codegen/function.rs [hunk 2]
```diff
@@ -82,6 +83,12 @@ pub fn generate(
...
```
Comment (GuillaumeGomez) [inline src/codegen/function.rs:89]: Actually: isn't it still generated doc alias we don't want? Do you have an example now please?

Comment (fengalin) [reply src/codegen/function.rs:89]: https://github.com/fengalin/gtk-rs/blob/309d221a728c5d706cbd553ebe96874a8ec2cba9/glib/src/auto/date_time.rs#L47

Comment (GuillaumeGomez) [reply src/codegen/function.rs:89]: I only see one `doc(alias = "...")` in this link. Did I miss something?

Comment (fengalin) [reply src/codegen/function.rs:89]: Isn't that what you wanted: https://github.com/gtk-rs/gir/pull/1014#discussion_r536735301?

Comment (fengalin) [reply src/codegen/function.rs:89]: For explicit renames in Gir.toml, you get this: https://github.com/fengalin/gtk-rs/blob/309d221a728c5d706cbd553ebe96874a8ec2cba9/gtk/src/auto/widget.rs#L340

Comment (GuillaumeGomez) [reply src/codegen/function.rs:89]: At yes, I meant the second, sorry, should have been more explicit. And once again, we fall into the same issue, it generates very common names such `get_name`. :-/

Comment (fengalin) [reply src/codegen/function.rs:89]: Yes, that might be the case with many names. I can remove this, but I thought that would help users figure out changes in names. This is particularly helpful for getters renaming (upcoming).

Comment (GuillaumeGomez) [reply src/codegen/function.rs:89]: I think it's not a good idea (even less if it's meant to be temporary). What do you think @sdroege ?

Comment (fengalin) [reply src/codegen/function.rs:89]: It's nice that we can serch by the C name, but too bad we don't get results by the actual function name, nor by searching something like "Widget name".

Comment (fengalin) [reply src/codegen/function.rs]: Removed

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me! Just waiting for @sdroege and we're good to go. :)

**Comment by fengalin** [comment]
Last minute inspiration... sorry.
Added a function so that we don't duplicate the `new_name` / `name` selection code in codegen.

**Comment by sdroege** [comment]
Please send the regenerate PRs for gtk-rs and gstreamer :)

**Comment by fengalin** [comment]
I'll update the PR I already pushed now, but please don't merge to master in those repositories while I update.

**Comment by fengalin** [comment]
Well I guess gtk-rs is already done by 0ef36307474678fdd11e09d995b28e5ae32e35d9.
So skipping this one.

**Pull request merged**